### PR TITLE
NFC Minor code cleanup to contribution pages

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Just some minor cleanups.

Before
----------------------------------------

After
----------------------------------------

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Civi\Financialacls\BuildAmountHookTest::testBuildAmount
Failed asserting that actual size 2 matches expected size 1.

/home/jenkins/bknix-dfl/build/core-19468-621c8/web/sites/all/modules/civicrm/ext/financialacls/tests/phpunit/Civi/Financialacls/BuildAmountHookTest.php:40
/home/jenkins/bknix-dfl/extern/phpunit6/phpunit6.phar:570

**Comment by mattwire** [comment]
test this please

File: CRM/Price/BAO/PriceSet.php [hunk 1]
```diff
@@ -871,13 +871,13 @@ public static function buildPriceSet(&$form) {
...
     if ($className == 'CRM_Contribute_Form_Contribution_Main') {
-      $feeBlock = &$form->_values['fee'];
+      $feeBlock = $form->_values['fee'];
       if (!empty($form->_useForMember)) {
         $component = 'membership';
       }
     }
     else {
-      $feeBlock = &$form->_priceSet['fields'];
+      $feeBlock = $form->_priceSet['fields'];
```
Comment (demeritcowboy) [inline CRM/Price/BAO/PriceSet.php:880]: I haven't traced this but if it helps - just looking at the code below if you take away the `&` here then the hook won't be able to update $feeBlock, so when it's passed to addPriceFieldsToForm it's still the original value.

Comment (demeritcowboy) [reply CRM/Price/BAO/PriceSet.php:880]: Correction: Won't be able to update the original `$form->_priceSet['fields']`.

Comment (mattwire) [reply CRM/Price/BAO/PriceSet.php]: Ah yes, thankyou. I just saw the `&` and thought PHP4..

Comment (mattwire) [reply CRM/Price/BAO/PriceSet.php]: Removed from this PR

File: CRM/Contribute/Form/ContributionBase.php [hunk 2]
```diff
@@ -207,6 +207,15 @@ class CRM_Contribute_Form_ContributionBase extends CRM_Core_Form {
...
 
+  /**
+   * The contribution ID
+   *
+   * @var int
+   *
+   * @fixme this is retrieved as a URL param. I think it is only set when using the contribution page in "invoice" mode
+   */
+  protected $_ccid;
+
```
Comment (demeritcowboy) [inline CRM/Contribute/Form/ContributionBase.php:218]: It's used publicly in Contribution_Main::formRule. BUT...

it seems like it's still able to access it - maybe a php thing I'm not aware of. I would assume the below would give an error, but it doesn't seem to:

```php
class A {
  protected $bar;
}
class B extends A {
  function do_something($z) {
    echo $z->bar; // It's accessing bar as a public member, not as $this->bar, but maybe php looks and sees it's a `B` and it knows we're inside `B`?
  }
}
$form = new B();
$form->do_something($form);
```

So anyway, the rest of the PR looks good.

Comment (eileenmcnaughton) [reply CRM/Contribute/Form/ContributionBase.php:218]: Yeah - I see it - adding a public getter would be one option

Screenshot from 2021-02-01 13-35-12

**Comment by eileenmcnaughton** [comment]
My understanding is that ccid in the url is an option if you are making a payment against an existing contribution (personally I'd rather see us expose the add payment form but it's complicated..... & the contribution pages are crazy overloaded in general)

File: CRM/Contribute/Form/ContributionBase.php (hunk 2 above)
Comment (mattwire) [reply CRM/Contribute/Form/ContributionBase.php:217]: Thanks for spotting the protected/public thing.

**Comment by mattwire** [comment]
@demeritcowboy @eileenmcnaughton Thanks for reviewing. I've changed `_ccid` to public. Adding a getter in this PR is a step too far as this was just about understanding what that param was rather than actually doing any cleanup/refactoring.

**Comment by eileenmcnaughton** [comment]
OK - looks good now

**Pull request merged**